usaspending_api/references/v2/views/autocomplete.py

```diff
@@ -99,6 +99,44 @@ def post(self, request):
         return Response(response)
 
 
+class CFDAAutocompleteViewSet(BaseAutocompleteViewSet):
+
+    def post(self, request):
+        """Return all budget function/subfunction titles matching the provided search text"""
+
+        search_text, limit = self.get_request_payload(request)
+
+        # get relevant TransactionContracts
+        queryset = Cfda.objects.filter(program_number__isnull=False,
+                                       program_title__isnull=False)
+        # Filter based on search text
+        response = {}
+
+        queryset = queryset.annotate(similarity=Greatest(
+            TrigramSimilarity('program_number', search_text),
+            TrigramSimilarity('program_title', search_text),
+            TrigramSimilarity('popular_name', search_text)))\
+            .distinct().order_by('-similarity')
+
+        queryset = queryset.annotate(prog_num_similarity=TrigramSimilarity('program_number', search_text))
+
+        cfda_exact_match_queryset = queryset.filter(similarity=1.0)
+        if cfda_exact_match_queryset.count() > 0:
+            # check for trigram error if similarity is a program number
+            if cfda_exact_match_queryset.filter(prog_num_similarity=1.0).count() > 0:
+                if len(cfda_exact_match_queryset.first().program_number) == len(search_text):
```

Comment:
this is checking for the above error. 

I am looking for false positives (100% similarity) on line 126